code/unclassified/src/apriori/apriori.py

```diff
@@ -0,0 +1,131 @@
+# Part of Cosmos by OpenGenus Foundation
+# Find frequently occuring subsets of items in a set of transactions
+# Contributed by: Aditya Lohana (dundermiflin)
+
+
+import csv
+from itertools import combinations, chain
+
+
+min_support = 0.1
+
+
+def csv_to_list(filename):
+	"""
+		Function to convert the transaction-dataset into a list of lists
+	"""
+	with open(filename, 'r') as f:
+		reader = csv.reader(f)
+		data = [row for row in reader]
+		return data
+
+
+def count_support(transactions):
+	"""
+		Function to count the support of each unique item in the dataset
+		Returns a dictionary with item names as keys, and support as values
+	"""
+	sup = {}
+	num_transactions = len(transactions)
+	for row in transactions:
+		for col in row:
+			if col not in sup.keys():
+				sup[col] = 1
+			else:
+				sup[col] += 1
+	for item in sup.keys():
+		sup[item] /= num_transactions
+	return sup
+
+
+def join_itemset(a, b):
+	"""
+		Function to merge to length-N itemsets with (N-1)-length common prefix
+		Returns an itemset of length N+1
+	"""
+	a = list(a)
+	b = list(b)
+	x = a[: -1]
```

Comment:
No space before negative sign.

Reply:
Thanks for pointing it out. I never paid attention before to proper styling while writing code.  Might take a while to get a hang of it!